doozerlib/image.py

```diff
@@ -657,6 +657,31 @@ def run_docker_cov(cmd):
 
             write_record()
 
+    def calculate_config_digest(self, group_config, streams):
+        image_config: Dict[str, Any] = self.config.primitive()  # primitive() should create a shallow clone for the underlying dict
+        group_config: Dict[str, Any] = group_config.primitive()
+        streams: Dict[str, Any] = streams.primitive()
+        del image_config["owners"]  # ignore the owners entry for the digest: https://issues.redhat.com/browse/ART-1080
+        message = {
+            "config": image_config,
+        }
+
+        repos = set(image_config.get("enabled_repos", []) + image_config.get("non_shipping_repos", []))
+        if repos:
+            message["repos"] = {repo: group_config["repos"][repo] for repo in repos}
+
+        builders = image_config.get("from", {}).get("builder", [])
+        from_stream = image_config.get("from", {}).get("stream")
+        referred_streams = {builder.get("stream") for builder in builders if builder.get("stream")}
+        if from_stream:
+            referred_streams.add(from_stream)
+        if referred_streams:
+            message["streams"] = {stream: streams[stream] for stream in referred_streams}
+
+        message = util.sort_dict(message)
+        digest = hashlib.sha256(json.dumps(message).encode("utf-8")).hexdigest()
```

Comment:
`sort_dict` is clever, but why not just do:
```suggestion
        digest = hashlib.sha256(json.dumps(message, sort_keys=True).encode("utf-8")).hexdigest()
```

Reply:
That's convenient. Didn't realize there is a simpler way :)